Paraview

DeyuChen opened the pull request:
The first version of the paraviewweb frontend.

Screenshot:
![screenshot from 2017-06-01 17-02-52](https://cloud.githubusercontent.com/assets/27230373/26705751/0ccdb97c-46ed-11e7-962f-5281af62512a.png)

Screenshot:
![screenshot from 2017-06-01 17-04-07](https://cloud.githubusercontent.com/assets/27230373/26705747/05487b60-46ed-11e7-9459-72ad20c4acac.png)

JavierJia commented:
@DeyuChen Can you add an HTTP request to register the `ds_tweet` dataset for the purpose of only running the paraview w/o twittermap? The corresponding code can be found here: https://github.com/ISG-ICS/cloudberry/blob/master/twittermap/app/model/Migration_20170428.scala

JavierJia commented on paraview/src/CloudberryDemo.js:
```diff
@@ -0,0 +1,437 @@
+/* global document */
```
is this file the same as `paraview/dist/CloudberryDemo.js`?

DeyuChen replied:
paraview/src/CloudberryDemo.js is the source code and paraview/dist/CloudberryDemo.js is the code that is generated by webpack, which combines the source code with all of the required libraries into one JavaScript file. The *.woff2 and *.ttf are generated by webpack, too. I leave these files there because I think it would be easier for people who just want to try it but does not want to modify the source code. I can delete them if they are not necessary.

JavierJia replied:
if that is the case, can you add the script that can generate this files instead?

HotLemonJuice commented on paraview/src/CloudberryDemo.js:
```diff
@@ -0,0 +1,437 @@
+        default:
+            break;
+    }
+    return 'good';
```
I think you should throw an Error here.

JavierJia commented:
@DeyuChen any updates on this PR? 😄

DeyuChen commented:
I just updated the PR. I throw an exception as @HotLemonJuice suggested. I also added a bash script to generate the combine javascript file and other required files. I need to install some npm packages needed by ParaviewWeb manually in the script because they are missing in the package they provided. I think those packages are supposed to be installed by "kw-web-suite". If ParaviewWeb updates their package, we can update our script accordingly.

JavierJia commented on paraview/readme.txt:
```diff
@@ -0,0 +1,27 @@
+    $ sbt "project neo" "run"
+
+5) start a php server for the demo under "cloudberry/paraview/dist" directory
+    $ php -S [IP address]:[port]
```
change it to `php -S localhost:8888` for the ease of the user to try in a copy-and-paste fashion?

JavierJia commented on paraview/readme.txt:
```diff
@@ -0,0 +1,27 @@
+5) start a php server for the demo under "cloudberry/paraview/dist" directory
+    $ php -S [IP address]:[port]
+
+The demo should be available at http://[IP address]:[port]/
```
same here.

DeyuChen replied:
Yes, the paraview/dist/CloudberryDemo.js is generated and I just removed it. I also updated the readme.

JavierJia commented on paraview/generate_dist_javascript.sh:
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+npm install paraviewweb  --save
```
I assume some of the lines will generate the file we removed, could you add some comments to explain what the purpose of each command?

JavierJia commented on paraview/src/CloudberryDemo.js:
```diff
@@ -0,0 +1,441 @@
+var JSONbig = require('json-bigint');
+
+var httpProxy = "proxy.php";
+var cloudberryHost = "ws://localhost:9000/ws"
```
I think we can add a comment to remind the user to specify this is the line to change.

DeyuChen replied:
I have added some comments in both files.

JavierJia commented:
You will need to pull the master. 
And there is one good review comments from "codacy" here: https://www.codacy.com/app/ISG-Cloudberry/cloudberry/pullRequest?prid=681063
